reconcile/utils/ldap_client.py

```diff
@@ -22,16 +24,13 @@ def init_from_config():
 
     serverUrl = config['ldap']['server']
     _base_dn = config['ldap']['base_dn']
-
     return init(serverUrl)
 
 
 def user_exists(username):
-    global _client
     global _base_dn
 
-    init_from_config()
-
-    search_filter = "uid={},{}".format(username, _base_dn)
-
-    return _client.search(search_filter, '(objectclass=person)')
+    with init_from_config() as client:
+        search_filter = "uid={},{}".format(username, _base_dn)
+        result, _, _, _ = client.search(search_filter, '(objectclass=person)')
```

Comment:
```suggestion
        result, _, _, _ = client.search(base_dn, f'(&(uid={username})(objectclass=person))')
```
because that way, you can move towards having far more `uid` s in a single filter, and reduce the amount of round-trips you do to LDAP in any integration.